### PR TITLE
Consolidate network components

### LINKED ISSUE #996: Consolidate network components into one
Dependency of #924 and #1038 

Currently our frontend separates the idea of a regular network (think hairball type) from a bipartite or other network. This is not ideal. It would be more helpful to have a single `Network` component that could handle all the cases. 

A little background:
- Originally we started abstracting away from a `Network` component becuase it seemed like it would end up just being visx's `Graph` component.
- We're realizing now that our component would differ a bit from the visx `Graph`. It would have default layouts and node interactions. 
- We're realizing we want all of our networks across the site to look and feel the same. That goal will be much more easily managed if we have only one `Network` component.

`Network` would be very similar to `BipartiteNetwork` but would be more general. For starters,
> The data prop should be "exploded" to be the props nodes, links, and partitions, the last being optional
The type of the partitions prop should be { name?: string, nodeList: string[]; }
If partitions is empty, then a simple graph will be rendered

( from Dave in slack)

`Network` should have a default layout if node x and y values are not provided. For the case where the graph is partitioned, the default layout would be nodes aligned in columns according to their partitions. For the case of no partitions, the default should be a ring.

### PR DESCRIPTION
**Pull request by asizemore**
Resolves #996 

The goal here is to simplify our network plot components. The approach I've taken here deviates slightly from the original proposal, mostly because cramming all the logic into NetworkPlot became overwhelming. So we'll see what others think!

Left to do:

- [x] Create `NetworkPlot`
- [x] Handle svg styling and overrides
- [x] Pass empty network data
- [x] Handle default layouts
- [x] Move partition titles into bp net styles
- [x] Possibly remove `partitions` from `NetworkPlot`. That prop is only used to set node positions, and if `BipartiteNetworkPlot` does that already...
- [x] Possibly setting a max height so that the network scrolls
- [x] Add more network stories

Left for later because they're more feature-upgrade types:
1. Allow `NodeWithLabel` to take a label position, rotation, etc.

Other notes:
- I'm not really happy with how the `svgStyles` handles the `containerSyles` height and width. It assumes the input is a number, and anything else provided gets the boot (the default boot, here). I don't have a much better way to handle this right now, but just wanted to state my frustration
- I moved the css around for sake of organization. I didn't change styles, just class names when appropriate.

### CODE REVIEW

**Comment by dmfalke** [comment]
@asizemore I will start to review this in the next day or two. Let me know if you want me to take a look sooner!

**Comment by asizemore** [comment]
No problem, thanks @dmfalke ! This isn't for b68 so can definitely wait

**Review by dmfalke — COMMENTED** [review]
I think this looks great. I like the direction you went with this.

Have you given any thought into how various layout algorithms will be applied, in the future? And would a bipartite network be considered a layout? I don't really have any strong opinions on this, but I think it's an interesting thing to consider.

File: packages/libs/components/src/plots/BipartiteNetworkPlot.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      }),
+    [
+      column1Position,
+      column2Position,
+      partitions,
+      nodesByPartition,
+      svgStyles.nodeSpacing,
+      svgStyles.topPadding,
+    ]
+  );
```
[the remark this answers is not in the document]
Comment (asizemore) [reply packages/libs/components/src/plots/BipartiteNetworkPlot.tsx]: Nice! Done and done

File: packages/libs/components/src/plots/NetworkPlot.tsx [hunk 2]
```diff
@@ -114,82 +102,62 @@ function BipartiteNetwork(
...
-      svgStyles.nodeSpacing,
-      svgStyles.topPadding,
-    ]
+      nodes.map((node, index) => ({
+        labelPosition: 'right' as LabelPosition,
+        ...node,
+        x: node.x ?? 230 + 200 * Math.cos(2 * Math.PI * (index / nodes.length)),
+        y: node.y ?? 230 + 200 * Math.sin(2 * Math.PI * (index / nodes.length)),
+        actions: getNodeActions?.(node.id),
+      })),
```
[the remark this answers is not in the document]
Comment (asizemore) [reply packages/libs/components/src/plots/NetworkPlot.tsx]: Skipped in the latest commits

**Comment by asizemore** [comment]
@dmfalke ready for another look!

I let `NetworkPlot` assume it's getting node coordinates, but to totally remove the traversal i also have it assuming it's receiving node actions as well. Not a bad thing since we can totally remove a loop over the nodes. I think when it comes time for the network plot, it's possible we create a `LayoutNetwork` or `LayoutedNetwork` or not sure how to say it yet :) but some component that does what `BipartiteNetworkPlot` does now that lays out the network and processes nodes and edges in a way that nicely preps them for `NetworkPlot`. 

Your point about the bipartite network being a layout is a good one. Mathematically it's much more than a layout because it carries a bunch of assumptions and rules about the network. But for this codebase, how much does that matter? Is it really anything more than a nice layout of the network when special data is given? I've been asking myself this a lot.

But anyways to get back to your question about layouts, yes I have given that a bunch of thought lately. Right now I'd like the backend to do it and to send x/y coordinates back to the frontend. I also want the backend to (someday) include node labels in that layout calculation and send us those coordinates as well so that we don't have a bunch of overlapping labels. The frontend could do a lot of layouts, but I _think_ the backend would be faster and more easily expanded to specialized layouts if we go that direction in the future.

**Comment by dmfalke** [comment]
This all sounds great!

**Review by dmfalke — APPROVED** [review]

**Pull request merged**